### PR TITLE
Subscription refactoring phase 1

### PR DESCRIPTION
**Pull request by federicocappelli**
Task/Issue URL: https://app.asana.com/0/72649045549333/1206805455884775/f
iOS PR: https://github.com/duckduckgo/iOS/pull/2818
macOS PR: https://github.com/duckduckgo/macos-browser/pull/2740
What kind of version bump will this require?: Major
Tech Design URL: https://app.asana.com/0/1205842942115003/1207147511614062/f

**Description**:

Subscription refactoring phase 1: https://app.asana.com/0/72649045549333/1207157941206685/f

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [ ] iOS 16
* [ ] macOS 10.15
* [ ] macOS 11
* [ ] macOS 12

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: Sources/Subscription/AccountManager.swift [hunk 1]
```diff
@@ -182,6 +207,10 @@ public class AccountManager: AccountManaging {
         NotificationCenter.default.post(name: .accountDidSignIn, object: self, userInfo: nil)
     }
 
+    public func signOut() {
+        signOut(skipNotification: false)
```
Comment (afterxleep) [inline Sources/Subscription/AccountManager.swift:211]: defaulting skipNotification to false on this method will make these calls cleaner.

File: Sources/Subscription/AccountManager.swift [hunk 2]
```diff
@@ -305,7 +339,7 @@ public class AccountManager: AccountManaging {
         }
     }
 
-    public typealias AccountDetails = (email: String?, externalID: String)
+//    public typealias AccountDetails = (email: String?, externalID: String)
```
Comment (afterxleep) [inline Sources/Subscription/AccountManager.swift:342]: Leftover?

**Review by afterxleep — APPROVED** [review]
LGTM.  Just 2 nitpicks

**Comment by federicocappelli** [comment]
Closing, I'll create PRs for a more advanced refactoring stage

**Pull request closed without merging**